src/components/__tests__/Calendar.spec.ts

```diff
@@ -0,0 +1,320 @@
+                    onEventDrop: jasmine.any(Function),
+                    onEventResize: jasmine.any(Function),
+                    onSelectEvent: jasmine.any(Function),
+                    onSelectSlot: jasmine.any(Function)
+                }),
+                createElement(Alert, { className: "widget-calendar-alert" })
+            )
+        );
+    });
+
+    it("renders events with custom colors", () => {
+        const calendarComponent = renderCalendar(calendarProps);
+        const style = { width: "100%", height: "580px" };
+        const events = { color: "red" };
+        (calendarComponent.instance() as any).eventColor(events);
+
+        expect(calendarComponent).toBeElement(
+            createElement("div", { className: "widget-calendar", style },
+                createElement(DragAndDropCalendar, {
+                    events: [],
+                    allDayAccessor: jasmine.any(Function),
+                    eventPropGetter: jasmine.any(Function),
+                    defaultView: "month",
+                    formats,
+                    messages: "",
+                    views: [],
+                    popup: true,
+                    selectable: true,
+                    step: 60,
+                    showMultiDayTimes: true,
+                    onEventDrop: jasmine.any(Function),
+                    onEventResize: jasmine.any(Function),
+                    onSelectEvent: jasmine.any(Function),
+                    onSelectSlot: jasmine.any(Function)
+                }),
+                createElement(Alert, { className: "widget-calendar-alert" })
+            )
+        );
+    });
+
+    describe("event handler", () => {
+        it("#onSelectSlot() calls the parent onSelectSlot handler", () => {
+            calendarProps.onSelectSlotAction = jasmine.createSpy("onClick");
+            const calendarComponent = renderCalendar(calendarProps);
+            (calendarComponent.instance() as any).onSelectSlot();
+
+            expect(calendarProps.onSelectSlotAction).toHaveBeenCalled();
+        });
+
+        it("#onSelectEvent() calls the parent onSelectEvent handler", () => {
+            calendarProps.onSelectEventAction = jasmine.createSpy("onClick");
+            const calendarComponent = renderCalendar(calendarProps);
+            (calendarComponent.instance() as any).onSelectEvent();
+
+            expect(calendarProps.onSelectEventAction).toHaveBeenCalled();
+        });
+
+        it("#onEventDrop() calls the parent onEventDrop handler when the dates are different", () => {
+            const eventInfo = {
+                start: new Date(),
+                event: {
+                    start: new Date((new Date()).valueOf() + 1000 * 3600 * 24)
+                }
+            };
+            calendarProps.onEventDropAction = jasmine.createSpy("onDrop");
+            const calendarComponent = renderCalendar(calendarProps);
+            (calendarComponent.instance() as any).onEventDrop(eventInfo);
```

Comment:
I suggest u declare a `calendarComponent.instance() as any` within a jasmine `beforeAll` function and re-use it instead....its over repeated.